### ISSUE TITLE
Segfault on shutdown (wayland)

### ISSUE THREAD

**Issue by 1ace** [alacritty/alacritty#4751]
### System

OS: ArchLinux + sway
Version: `alacritty 0.7.1 (a852eb85d57a9c966cea)`

### Logs

As the title says, at the time of alacritty's shutdown, it crashes:
```
Thread 10 "smithay-clipboa" received signal SIGSEGV, Segmentation fault.
#0  0x00007ffff7fbfa30 in wl_list_insert () from /usr/lib/libwayland-client.so
#1  0x00007ffff7fbc8ff in wl_display_read_events () from /usr/lib/libwayland-client.so
#2  0x00007ffff7fbcf54 in wl_display_dispatch_queue () from /usr/lib/libwayland-client.so
#3  0x00007ffff7fbd1e0 in wl_display_roundtrip_queue () from /usr/lib/libwayland-client.so
#4  0x0000555555a8da5e in wayland_client::imp::event_queue::{{impl}}::sync_roundtrip::{{closure}}<closure-5> () at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.28.3/src/native_lib/event_queue.rs:85
#5  0x0000555555a8f026 in wayland_client::imp::event_queue::with_dispatch_meta::{{closure}}<core::result::Result<u32, std::io::error::Error>,closure-5,closure-0> () at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.28.3/src/native_lib/event_queue.rs:24
#6  0x0000555555a603df in scoped_tls::ScopedKey<core::cell::RefCell<(&mut FnMut<(wayland_client::RawEvent, wayland_client::proxy::Main<wayland_client::anonymous_object::AnonymousObject>, wayland_commons::filter::DispatchData)>, wayland_commons::filter::DispatchData)>>::set<core::cell::RefCell<(&mut FnMut<(wayland_client::RawEvent, wayland_client::proxy::Main<wayland_client::anonymous_object::AnonymousObject>, wayland_commons::filter::DispatchData)>, wayland_commons::filter::DispatchData)>,closure-0,core::result::Result<u32, std::io::error::Error>> (self=0x55555668a4d0 <wayland_client::imp::event_queue::DISPATCH_METADATA>, t=0x7fffe4379918, f=...) at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
#7  0x0000555555a8ef2f in wayland_client::imp::event_queue::with_dispatch_meta<core::result::Result<u32, std::io::error::Error>,closure-5,closure-0> (fb=..., data=..., f=...) at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.28.3/src/native_lib/event_queue.rs:24
#8  0x0000555555a8d48f in wayland_client::imp::event_queue::EventQueueInner::sync_roundtrip<closure-5> (self=0x7fffc0000c40, data=..., fallback=...) at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.28.3/src/native_lib/event_queue.rs:83
#9  0x0000555555a3fb5b in wayland_client::event_queue::EventQueue::sync_roundtrip<smithay_clipboard::worker::dispatch_data::ClipboardDispatchData,closure-5> (self=0x7fffe4379c28, data=0x7fffe437a368, fallback=...) at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.28.3/src/event_queue.rs:191
#10 0x0000555555a927c4 in smithay_clipboard::worker::worker_impl (display=..., request_rx=..., reply_tx=...) at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-clipboard-0.6.2/src/worker/mod.rs:319
#11 0x0000555555a90f4c in smithay_clipboard::worker::spawn::{{closure}} () at /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-clipboard-0.6.2/src/worker/mod.rs:44
#12 0x0000555555a4a682 in std::sys_common::backtrace::__rust_begin_short_backtrace<closure-0,()> (f=...) at /build/rust/src/rustc-1.49.0-src/library/std/src/sys_common/backtrace.rs:125
#13 0x0000555555a5a0e1 in std::thread::{{impl}}::spawn_unchecked::{{closure}}::{{closure}}<closure-0,()> () at /build/rust/src/rustc-1.49.0-src/library/std/src/thread/mod.rs:474
#14 0x0000555555a5fbc1 in std::panic::{{impl}}::call_once<(),closure-0> (self=..., _args=()) at /build/rust/src/rustc-1.49.0-src/library/std/src/panic.rs:322
#15 0x0000555555a67649 in std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,()> (data=0x7fffe437aa48) at /build/rust/src/rustc-1.49.0-src/library/std/src/panicking.rs:381
#16 0x0000555555a680ed in __rust_try ()
#17 0x0000555555a67574 in std::panicking::try<(),std::panic::AssertUnwindSafe<closure-0>> (f=...) at /build/rust/src/rustc-1.49.0-src/library/std/src/panicking.rs:345
#18 0x0000555555a67433 in std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,()> (f=...) at /build/rust/src/rustc-1.49.0-src/library/std/src/panic.rs:396
#19 0x0000555555a59eac in std::thread::{{impl}}::spawn_unchecked::{{closure}}<closure-0,()> () at /build/rust/src/rustc-1.49.0-src/library/std/src/thread/mod.rs:473
#20 0x0000555555a83b0f in core::ops::function::FnOnce::call_once<closure-0,()> () at /build/rust/src/rustc-1.49.0-src/library/core/src/ops/function.rs:227
#21 0x00005555562aee5a in alloc::boxed::{{impl}}::call_once<(),FnOnce<()>,alloc::alloc::Global> (self=..., args=<optimized out>) at /build/rust/src/rustc-1.49.0-src/library/alloc/src/boxed.rs:1307
#22 alloc::boxed::{{impl}}::call_once<(),alloc::boxed::Box<FnOnce<()>, alloc::alloc::Global>,alloc::alloc::Global> (self=0x5555570b5f80, args=<optimized out>) at /build/rust/src/rustc-1.49.0-src/library/alloc/src/boxed.rs:1307
#23 std::sys::unix::thread::{{impl}}::new::thread_start (main=0x5555570b5f80) at library/std/src/sys/unix/thread.rs:71
#24 0x00007ffff7dce3e9 in start_thread () from /usr/lib/libpthread.so.0
#25 0x00007ffff7bb4293 in clone () from /usr/lib/libc.so.6
```

Possibly something to report to `smithay-clipboard` instead, but I chose to report it here instead in case it's `alacritty` doing something wrong, as I have a feeling it's closing the wayland connection before joining the `smithay-clipboard` thread.

This is 100% reproducible on my machine; I'm happy to help debug or test patches :)

**Comment by nixpulvis** [comment]
This issue was recently fixed in #4742. Closing as dup.

**Comment by 1ace** [comment]
Ah awesome, thanks @chrisduerr and @nixpulvis !
